src/containers/Main/AuthenticatedWrapper.tsx

```diff
@@ -4,25 +4,35 @@ import { useRouter } from '../../utils/hooks/useRouter'
 import isLoggedIn from '../../utils/helpers/loginCheck'
 import NonRegisteredLogin from '../User/NonRegisteredLogin'
 import { Redirect } from 'react-router'
-import { Loading } from '../../components'
 
 const AuthenticatedContent: React.FC = () => {
-  const {
-    location,
-    query: { sessionId },
-  } = useRouter()
-  // Location is not fully loaded unless 'state' is present, TODO try upgrading react router
-  // without this sessionId is missing when navigating via Click Here in forgot password verification
-  if ('state' in location) {
-    let { pathname, search } = location
-    // If there is a sessionId in the URL, then need to login as nonRegistered
-    // before continuing
-    if (sessionId) return <NonRegisteredLogin option="redirect" redirect={pathname + search} />
-    if (isLoggedIn()) return <SiteLayout />
-    return <Redirect to={{ pathname: '/login', state: { from: location.pathname } }} />
-  }
-
-  return <Loading />
+  const { location } = useRouter()
+
+  const sessionId = getSessionIdFromUrl()
+
+  const { pathname, search } = location
+
+  // If there is a sessionId in the URL, then need to login as nonRegistered
+  // before continuing
+  if (sessionId && !isLoggedIn())
+    return <NonRegisteredLogin option="redirect" redirect={pathname + search} />
+
+  if (isLoggedIn()) return <SiteLayout />
+
+  return <Redirect to={{ pathname: '/login', state: { from: location.pathname } }} />
 }
 
 export default AuthenticatedContent
+
+// This is a nasty hack that we need due to the fact that ReactRouter doesn't
+// have "query" values available when this component is loaded from another
+// (i.e. when clicking "Click here" in password reset verification). So we fetch
+// and parse it ourselves directly from window.location.search instead. We
+// should upgrade to ReactRouter 6 at some point and hopefully this will be
+// sorted.
+export const getSessionIdFromUrl = () => {
+  const sessionIdRegex = /^\?(sessionId)=(.+)$/
+  const matches = window.location.search.match(sessionIdRegex)
```

Comment:
I would have used query-string (it's already in our dependencies)

Reply:
Oh okay. Yeah, I'll change it.

Reply:
Okay, I think that this whole "missing sessionId" thing is actually a bit of a red herring.

Explanation and new PR here: #1285 

Hopefully that'll be the last of it for a while. 🙄